### PR TITLE
Add "order" configuration and some optimize

### LINKED ISSUE #1: Maybe the execution life cycle is wrong?
I can write like this when using "vite-plugin-html", but I get an error when using "vite-plugin-simple-html".

image

So I looked at your implementation and adjusted it to:
image

Still not working，Maybe the execution life cycle is wrong?

### PR DESCRIPTION
**Pull request by Xdy1579883916**
Add "order" configuration, optimize EJS replacement, optimize test cases - close #1

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -22,64 +23,67 @@ const defaultMinifyOptions: HtmlMinifierTerserOptions = {
...
-      // Replace ejs variables
-      if (data) {
-        Object.entries(data).forEach(([key, value]) => {
-          result = result.replaceAll(new RegExp(`<%= ${key} %>`, 'g'), value);
-        });
-      }
+        // Replace ejs variables
+        if (data) {
+          Object.entries(data).forEach(([key, value]) => {
+            result = result.replaceAll(new RegExp(`<%[=-] ${key} %>`, 'g'), value);
```
Comment (wojtekmaj) [inline src/index.ts:39]: I wonder if that's really the intended behavior. `<%=` is for replacing single variables, while `<%-` is more for... Embedding stuff?

File: src/index.ts [hunk 2]
```diff
@@ -22,64 +23,67 @@ const defaultMinifyOptions: HtmlMinifierTerserOptions = {
   minifyCSS: true,
 };
 
-export default function simpleHtmlPlugin({ minify: minifyOptions = false, inject }: Options = {}) {
+export default function simpleHtmlPlugin({ minify: minifyOptions = false, inject, order = "pre" }: Options = {}) {
```
Comment (wojtekmaj) [inline src/index.ts:26]: I wonder if we even need this option? Is there a good reason to have it? I'd love for this plugin to leave up to its name.

File: src/index.ts (hunk 1 above)
Comment (Xdy1579883916) [reply src/index.ts:39]: This is not that important to me. I can change my writing style to adapt to this plug-in. What really hinders me from using it is the following problem. If it cannot be handled before vite parse HTML, the plug-in cannot execute normally in my scenario.

**Comment by Xdy1579883916** [comment]
Thank you for approving some of my modification suggestions.

I see you have some new commits, when are you planning to publish them.

**Comment by wojtekmaj** [comment]
Thanks for your contribution. I marked you as a co-author of these improvements I pushed.

> when are you planning to publish them.

Now :)

**Pull request closed without merging**